src/tests/test_base_tensorflow_model_block.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import os
+import tensorflow as tf
+from pathlib import Path
+from unittest import TestCase
+from hypergol.base_tensorflow_model_block import BaseTensorflowModelBlock
+tf.config.experimental.set_visible_devices([], 'GPU')
+
+
+class ModelBlockExample(BaseTensorflowModelBlock):
+
+    def __init__(self, exampleEmbeddingSize, **kwargs):
+        super(ModelBlockExample, self).__init__(**kwargs)
+        self.exampleEmbeddingSize = exampleEmbeddingSize
+        self.softmaxLayer = None
+
+    def build(self, inputs_shape):
+        self.softmaxLayer = tf.keras.layers.Softmax(axis=-1)
+
+    def call(self, inputs, **kwargs):
+        return self.softmaxLayer(inputs)
+
+
+class TestBaseTensorflowModelBlock(TestCase):
+
+    def __init__(self, methodName='runTest'):
+        super(TestBaseTensorflowModelBlock, self).__init__(methodName=methodName)
+        self.location = 'test_tensorflow_model_block'
+        self.exampleEmbeddingSize = 1
+        self.exampleLogits = tf.constant([2, 3, 4], dtype=tf.float32)
+        self.expectedOutput = np.array([0.09003057, 0.24472848, 0.66524094], dtype=np.float32)
+
+    def setUp(self):
+        super().setUp()
+        Path(self.location).mkdir()
+        self.block = ModelBlockExample(exampleEmbeddingSize=self.exampleEmbeddingSize)
+        self.blockSaveFile = f'{self.location}/{self.block.get_name()}.json'
+
+    def tearDown(self):
+        super().tearDown()
+        try:
+            os.remove(self.blockSaveFile)
+        except FileNotFoundError:
+            pass
+        os.rmdir(self.location)
```

Comment:
this can mess things up if the test fails leaves something there and the next one bumps into it, can you check how other tests doing it?

there is a function called delete_if_exists, you shouldn't use exceptions in a teardown as you might catch something that is an actual problem. (lol just checked I do use that in `clean_directories` but on directories (ehh, we discuss this tomorrow on the meeting)

Reply:
Ah yeah.  I had started using shutil.rmtree() instead in the other tests.  Good spot!